<!--
GENERATED FILE - DO NOT EDIT
This file was generated by [MarkdownSnippets](https://github.com/SimonCropp/MarkdownSnippets).
Source File: /doc/mdsource/TestingExceptions.source.md
To change this file edit the source file and then execute ./run_markdown_templates.sh.
-->

<a id="top"></a>

# Testing Exceptions

<!-- toc -->
## Contents

  * [Testing exception messages](#testing-exception-messages)<!-- endtoc -->



## Testing exception messages

Testing exceptions with Approval Tests is very easy. Simply pass in a call to the function (usually wrapped in a lambda). Approval tests will execute the code, catch the exception, and verify the exception's message, i.e. `exception.what()`.

The exception thrown must inherit `std::exception`.

<!-- snippet: verify_exception_message_example -->
<a id='snippet-verify_exception_message_example'/></a>
```cpp
Approvals::verifyExceptionMessage([]() { /* your code goes here */ });
```
<sup><a href='/tests/DocTest_Tests/ApprovalsTests.cpp#L105-L107' title='File snippet `verify_exception_message_example` was extracted from'>snippet source</a> | <a href='#snippet-verify_exception_message_example' title='Navigate to start of snippet `verify_exception_message_example`'>anchor</a></sup>
<!-- endsnippet -->

---

[Back to User Guide](/doc/README.md#top)
